### DISCUSSION TITLE
Vscode format theme support?

### DISCUSSION THREAD

**Discussion by tony-sol** [sharkdp/bat#2883] [Q&A]
Is there a way to enable vscode theme format support for bat?
Now, it uses sublime's `.tmTheme` format, but IMHO it's quite obsolete - sublime now itself supports json format for themes/colorschemes.
Or maybe is there a tool to convert json theme (vscode or sublime) to tmTheme format?

**Comment by wellkilo** [comment]
bat cannot load VS Code JSON themes directly. Its highlighter is syntect, and current custom themes must be TextMate .tmTheme files; newer .sublime-color-scheme files are not supported either.

The supported workflow is:

    mkdir -p "$(bat --config-dir)/themes"
    cp MyTheme.tmTheme "$(bat --config-dir)/themes/"
    bat cache --build
    bat --list-themes
    bat --theme MyTheme file.rs

This limitation is explicit in bat's [Adding new themes](https://github.com/sharkdp/bat/blob/master/README.md#adding-new-themes) documentation.

For a VS Code theme, use a converter/exporter that produces a TextMate plist theme, then inspect the result before caching it. Conversion is not always exact: VS Code semantic-token rules and some workbench colors have no equivalent in bat's syntax-highlighting theme, so only tokenColors/TextMate scopes can be mapped reliably. If the theme repository also ships a .tmTheme variant, prefer that over an automated conversion.

After conversion, bat cache --build is required; simply dropping the JSON theme into the folder will not make it appear.
